### DISCUSSION TITLE
Stuck in the middle of ADS model process

### DISCUSSION THREAD

**Discussion by Benson-Hsueh** [neuropoly/axondeepseg#775] [Help needed] [answered]
Hi,

I have installed the AxonDeepSeg you developed on my Macbook Pro and my iOS system is macOS Monterey (ver. 12.7.1).
I can analyze the sample TEM images you provided on the website successfully, but not for TEM images from our lab.
I attached the description shown in the terminal window when I ran the ADS model process and the TEM images I used for the analysis. 
Any suggestions will be helpful.
We would like to use this axon analysis for our future studies.
Thank you in advance.

All the best,
Benson

<img width="1169" alt="0 0278 um-pixel" src="https://github.com/axondeepseg/axondeepseg/assets/152512025/f7dca274-eb76-4f08-a493-9957fd2a713b">
<img width="855" alt="Description in terminal window" src="https://github.com/axondeepseg/axondeepseg/assets/152512025/9ecf7fcd-a2a7-4f3b-a0c1-c602e988b559">
<img width="993" alt="0 0327 um-pixel" src="https://github.com/axondeepseg/axondeepseg/assets/152512025/3d5883c1-1c09-402d-978b-529bf358edbb">
[TEM images.zip](https://github.com/axondeepseg/axondeepseg/files/13516531/TEM.images.zip)

**Comment by Benson-Hsueh** [comment]
The TEM images are in the zip file. It's strange that I can not attach those Tiff files individually so I compress them to zip file.

**Comment by mathieuboudreau** [comment, marked as answer]
Hi @Benson-Hsueh  - it seems that because your resolution is an order of magnitude different from the resolution of the TEM training images we used, that the software is dividing your image into a lot of resampled "patches" (~224). You can adjust this type of thing with the "Zoom factor" in our settings, I just tried with a Zoom factor of 0.1 and it seems to have segmented your images decently well (you can adjust the options to see if you can do better; Zoom of 0.1 gave me 4 patches to segment, so increasing it slightly might improve things a a bit).
![Screenshot 2023-11-30 at 3 11 44 PM](https://github.com/axondeepseg/axondeepseg/assets/1421029/829d582a-4be4-4ae7-8965-cf0300bb5a0a)


As for your other image that you sent, the contrast between myelin and axon (and background) is very low (everything is bright), so either there was a staining issue or microscope issue there, it would be difficult to segment that type of image using the software without some manual image editing on your end to improve the contrast before trying our tool.

Hope this helps!

**Reply by Benson-Hsueh** [reply]
Dear @mathieuboudreau thanks for your reply. Is there any advice for the zoom factor? For example, if we are using 40X or 60X magnification under TEM to acquire the images, how much zoom factor do you suggest us for the analysis?
Thank you!

**Reply by hermancollin** [reply]
Hi @Benson-Hsueh. [Here](https://axondeepseg.readthedocs.io/en/latest/documentation.html#segment-images-using-a-zoom-factor), there are some tips on how to use the zoom factor and the sweep to get the best results.

As mentioned by @mathieuboudreau, I'm pretty sure you could enhance your segmentations just by enhancing the contrast on your images. This [blogpost](https://saiwa.ai/blog/image-contrast-enhancement/) talks about the basics for image contrast enhancement.
